### PR TITLE
[WIP] Reword Service Accounts docs

### PR DESCRIPTION
**Pull request by sftim**
Reword:
- https://kubernetes.io/docs/tasks/configure-pod-container/configure-service-account/
- https://kubernetes.io/docs/reference/access-authn-authz/service-accounts-admin/

### CODE REVIEW

File: content/en/docs/tasks/configure-pod-container/configure-service-account.md [hunk 1]
```diff
@@ -9,22 +9,24 @@ weight: 90
...
 
 {{% capture overview %}}
-A service account provides an identity for processes that run in a Pod.
+A _service account_ provides an identity for processes that run in a Pod.
 
-*This is a user introduction to Service Accounts. See also the
-[Cluster Admin Guide to Service Accounts](/docs/reference/access-authn-authz/service-accounts-admin/).*
+*This is a user introduction to Service Accounts. See also [Managing Service Accounts](/docs/reference/access-authn-authz/service-accounts-admin/), aimed at cluster operators.*
+
+When you, a human, access the cluster&mdash;for example, using `kubectl`)&mdash;you are
```
Comment (tengqm) [inline content/en/docs/tasks/configure-pod-container/configure-service-account.md]: - there is a bracket mismatch;
- I don't think we prefer any dashes in the text.

File: content/en/docs/reference/access-authn-authz/service-accounts-admin.md [hunk 2]
```diff
@@ -10,99 +10,107 @@ weight: 50
 ---
 
 {{% capture overview %}}
-This is a Cluster Administrator guide to service accounts. It assumes knowledge of
+This is a Cluster Operator guide to service accounts. It assumes knowledge of
```
[the remark this answers is not in the document]
Comment (sftim) [reply content/en/docs/reference/access-authn-authz/service-accounts-admin.md]: I've borrowed this wording from https://kubernetes.io/docs/reference/glossary/?all=true#term-cluster-operator

Maybe it doesn't make sense to have two separate documents explicitly aimed at people with different experience levels. Rather, we could have a detailed Reference-section guide and also a simpler Task guide.

Something for a future PR? I want to keep this one focused.

File: content/en/docs/reference/access-authn-authz/service-accounts-admin.md [hunk 3]
```diff
@@ -113,6 +121,9 @@ kubectl delete secret mysecretname
 
 ### Service Account Controller
 
-Service Account Controller manages ServiceAccount inside namespaces, and ensures
-a ServiceAccount named "default" exists in every active namespace.
+The service account controller manages ServiceAccount resources inside namespaces, and ensures
+that a ServiceAccount named `default` exists in every active namespace.
+{{% /capture %}}
+{{% capture whatsnext %}}
+* Read more details about [projected volumes](/docs/tasks/configure-pod-container/configure-projected-volume-storage/).
```
[the remark this answers is not in the document]
Comment (sftim) [reply content/en/docs/reference/access-authn-authz/service-accounts-admin.md]: Noted, and tweaked.

File: content/en/docs/reference/access-authn-authz/service-accounts-admin.md [hunk 4]
```diff
@@ -10,99 +10,107 @@ weight: 50
...
+  1. If the Pod does not contain any `ImagePullSecrets`, then `ImagePullSecrets` of the `ServiceAccount` are added to the pod.
+  1. It adds a `volume` to the pod that will contain a token for API access.
   1. It adds a `volumeSource` to each container of the pod mounted at `/var/run/secrets/kubernetes.io/serviceaccount`.
 
 Starting from v1.13, you can migrate a service account volume to a projected volume when
-the `BoundServiceAccountTokenVolume` feature gate is enabled.
-The service account token will expire after 1 hour or the pod is deleted. See more details about [projected volume](/docs/tasks/configure-pod-container/configure-projected-volume-storage/).
+the `BoundServiceAccountTokenVolume` [feature gate](/docs/reference/command-line-tools-reference/feature-gates/)
+is enabled.
+The service account token will expire after the Pod is deleted or after 60 minutes, whichever comes first.
```
[the remark this answers is not in the document]
Comment (sftim) [reply content/en/docs/reference/access-authn-authz/service-accounts-admin.md]: Noted, and tweaked.

File: content/en/docs/reference/access-authn-authz/service-accounts-admin.md [hunk 5]
```diff
@@ -10,99 +10,107 @@ weight: 50
...
 (and it is by default on most distributions), then it does the following when a pod is created or modified:
 
-  1. If the pod does not have a `ServiceAccount` set, it sets the `ServiceAccount` to `default`.
-  1. It ensures that the `ServiceAccount` referenced by the pod exists, and otherwise rejects it.
-  1. If the pod does not contain any `ImagePullSecrets`, then `ImagePullSecrets` of the `ServiceAccount` are added to the pod.
-  1. It adds a `volume` to the pod which contains a token for API access.
+  1. If the Pod does not have a `ServiceAccount` set, it sets the `ServiceAccount` to `default`.
+  1. It ensures that the `ServiceAccount` referenced by the Pod exists, and otherwise rejects it.
+  1. If the Pod does not contain any `ImagePullSecrets`, then `ImagePullSecrets` of the `ServiceAccount` are added to the pod.
+  1. It adds a `volume` to the pod that will contain a token for API access.
```
[the remark this answers is not in the document]
Comment (sftim) [reply content/en/docs/reference/access-authn-authz/service-accounts-admin.md]: Reworded

File: content/en/docs/reference/access-authn-authz/service-accounts-admin.md [hunk 6]
```diff
@@ -10,99 +10,107 @@ weight: 50
...
+    specific tasks (and enabling least-privilege access for performing those tasks).
   - Auditing considerations for humans and service accounts may differ.
-  - A config bundle for a complex system may include definition of various service
-    accounts for components of that system.  Because service accounts can be created
-    ad-hoc and have namespaced names, such config is portable.
+  - A configuration bundle for a complex workload may include definition of various
+    service accounts for different system components. Because service accounts can be
+    created ad-hoc and have namespaced names, such configuration is portable.
+    (If service account names were global, you could find that two namespaces both
+    wanted to use the same name for different service accounts.
```
[the remark this answers is not in the document]
Comment (sftim) [reply content/en/docs/reference/access-authn-authz/service-accounts-admin.md]: Snipped. Thanks.

File: content/en/docs/tasks/configure-pod-container/configure-service-account.md [hunk 7]
```diff
@@ -21,10 +21,13 @@ customized the behavior in your cluster, in which case this documentation may
 not apply.
 {{< /note >}}
 
-When you (a human) access the cluster (for example, using `kubectl`), you are
-authenticated by the apiserver as a particular User Account (currently this is
-usually `admin`, unless your cluster administrator has customized your cluster). Processes in containers inside pods can also contact the apiserver.
-When they do, they are authenticated as a particular Service Account (for example, `default`).
+When you, a human, access the cluster (for example, using `kubectl`) you are
+authenticated by the apiserver as a particular user identity. For example, you
```
Comment (kbhawkey) [inline content/en/docs/tasks/configure-pod-container/configure-service-account.md]: - Do you need, `a human`?
When you access a cluster using a command line tool, such as `kubectl`, the API server authenticates the request using a specific user account (currently ... ).
- user identity vs user account, what is the difference?

File: content/en/docs/tasks/configure-pod-container/configure-service-account.md [hunk 8]
```diff
@@ -21,10 +21,13 @@ customized the behavior in your cluster, in which case this documentation may
...
+might have authenticated as username `admin`.
+
+Processes in containers inside pods can also contact the apiserver.
+When they do, those processes are able to authenticate as a particular Service
```
Comment (kbhawkey) [inline content/en/docs/tasks/configure-pod-container/configure-service-account.md]: capitalize Pods?

File: content/en/docs/tasks/configure-pod-container/configure-service-account.md [hunk 9]
```diff
@@ -9,7 +9,7 @@ weight: 90
 ---
 
 {{% capture overview %}}
-A service account provides an identity for processes that run in a Pod.
+A _service account_ provides an identity for processes that run in a Pod.
```
Comment (kbhawkey) [inline content/en/docs/tasks/configure-pod-container/configure-service-account.md]: Service account vs service account?

File: content/en/docs/tasks/configure-pod-container/configure-service-account.md [hunk 10]
```diff
@@ -37,16 +40,18 @@ When they do, they are authenticated as a particular Service Account (for exampl
 
 {{% capture steps %}}
 
-## Use the Default Service Account to access the API server.
+## Use the default Service Account to access the API server
+
+Each namespace has a default ServiceAccount, named `default`.
```
Comment (kbhawkey) [inline content/en/docs/tasks/configure-pod-container/configure-service-account.md]: `ServiceAccount` is the resource?

File: content/en/docs/tasks/configure-pod-container/configure-service-account.md (hunk 7 above)
Comment (sftim) [reply content/en/docs/tasks/configure-pod-container/configure-service-account.md]: - dropped “a human”
- I'm using the term “identity” and the page doesn't mention “user account” anywhere anymore (unless I missed one)

File: content/en/docs/tasks/configure-pod-container/configure-service-account.md (hunk 8 above)
Comment (sftim) [reply content/en/docs/tasks/configure-pod-container/configure-service-account.md]: Adjusted

File: content/en/docs/tasks/configure-pod-container/configure-service-account.md (hunk 9 above)
Comment (sftim) [reply content/en/docs/tasks/configure-pod-container/configure-service-account.md]: I think “service account” is fine for English grammar. The Kubernetes resource is ServiceAccount, which gets introduced below.

File: content/en/docs/tasks/configure-pod-container/configure-service-account.md (hunk 10 above)
Comment (sftim) [reply content/en/docs/tasks/configure-pod-container/configure-service-account.md]: Yep.

**Comment by kbhawkey** [comment]
https://deploy-preview-14681--kubernetes-io-master-staging.netlify.com/docs/reference/access-authn-authz/service-accounts-admin/

**Comment by sftim** [comment]
I'd left “user account” in a heading. Fixed.
(this will break inbound links to that fragment, but as it's at the top of the page I'm not worried)

File: content/en/docs/tasks/configure-pod-container/configure-service-account.md (hunk 9 above)
Comment (kbhawkey) [reply content/en/docs/tasks/configure-pod-container/configure-service-account.md]: Do you need italics? Check the style guide.

File: content/en/docs/tasks/configure-pod-container/configure-service-account.md (hunk 10 above)
Comment (kbhawkey) [reply content/en/docs/tasks/configure-pod-container/configure-service-account.md]: OK. Check the heading.

File: content/en/docs/tasks/configure-pod-container/configure-service-account.md [hunk 11]
```diff
@@ -70,15 +76,17 @@ spec:
   ...
 ```
 
-The pod spec takes precedence over the service account if both specify a `automountServiceAccountToken` value.
+If both the ServiceAccount and the pod spec specify a value for
+`automountServiceAccountToken`, the pod spec takes precedence.
```
Comment (kbhawkey) [inline content/en/docs/tasks/configure-pod-container/configure-service-account.md]: PodSpec is the object

File: content/en/docs/tasks/configure-pod-container/configure-service-account.md [hunk 12]
```diff
@@ -245,32 +257,25 @@ imagePullSecrets:
...
 {{< feature-state for_k8s_version="v1.12" state="beta" >}}
 
 {{< note >}}
-This ServiceAccountTokenVolumeProjection is __beta__ in 1.12 and
-enabled by passing all of the following flags to the API server:
+ServiceAccountToken volume projection is __beta__ since Kubernetes v1.12,
+controlled by the `TokenRequestProjection`
+[feature gate](/docs/reference/command-line-tools-reference/feature-gates/).
+
+To use it, you will need to pass all of the following flags to the API server:
```
Comment (kbhawkey) [inline content/en/docs/tasks/configure-pod-container/configure-service-account.md]: - clean up note shortcode: ServiceAccountToken volume projection is controlled by the `TokenRequestProjection` ...
- To use this feature, pass the following flags to the API server:

File: content/en/docs/reference/access-authn-authz/service-accounts-admin.md [hunk 13]
```diff
@@ -10,79 +10,84 @@ weight: 50
...
-  - A config bundle for a complex system may include definition of various service
-    accounts for components of that system.  Because service accounts can be created
-    ad-hoc and have namespaced names, such config is portable.
+## Users vs service accounts
+
+Kubernetes [distinguishes](/docs/reference/access-authn-authz/authentication/#users-in-kubernetes)
+between the concept of a user and a service account for a number of reasons:
+
+  - Users represent humans. Service accounts are for processes, which
+    run in {{< glossary_tooltip text="pods" term_id="pod" >}}.
```
Comment (kbhawkey) [inline content/en/docs/reference/access-authn-authz/service-accounts-admin.md]: - I'm not sure this change helps clarify a user account vs. service account.
- are italics needed?

**Comment by tengqm** [comment]
/lgtm
When other feedbacks are addressed, we can move this in.

File: content/en/docs/tasks/configure-pod-container/configure-service-account.md (hunk 9 above)
Comment (sftim) [reply content/en/docs/tasks/configure-pod-container/configure-service-account.md]: Tweaked

File: content/en/docs/tasks/configure-pod-container/configure-service-account.md (hunk 11 above)
Comment (sftim) [reply content/en/docs/tasks/configure-pod-container/configure-service-account.md]: From a cluster operator's point of view, Pods have specifications. Pods are Kubernetes objects.
Maybe “the Pod's specification” or “the Pod's `.spec`”?

**Comment by sftim** [comment]
@tengqm feedback addressed _I think_.
If I've missed anything, my apologies - please let me know where.

File: content/en/docs/tasks/configure-pod-container/configure-service-account.md (hunk 11 above)
Comment (kbhawkey) [reply content/en/docs/tasks/configure-pod-container/configure-service-account.md]: OK

File: content/en/docs/reference/access-authn-authz/service-accounts-admin.md [hunk 14]
```diff
@@ -4,115 +4,142 @@ reviewers:
...
-Three separate components cooperate to implement the automation around service accounts:
-
-  - A Service account admission controller
-  - A Token controller
-  - A Service account controller
-
-### Service Account Admission Controller
-
-The modification of pods is implemented via a plugin
-called an [Admission Controller](/docs/reference/access-authn-authz/admission-controllers/). It is part of the apiserver.
```
Comment (kbhawkey) [inline content/en/docs/reference/access-authn-authz/service-accounts-admin.md]: @sftim, what happens to the controller info?

File: content/en/docs/tasks/configure-pod-container/configure-service-account.md [hunk 15]
```diff
@@ -226,15 +259,16 @@ secrets:
 - name: default-token-uudge
 ```
 
-Using your editor of choice (for example `vi`), open the `sa.yaml` file, delete line with key `resourceVersion`, add lines with `imagePullSecrets:` and save.
+Using your editor, delete the line with key `resourceVersion`, add lines for `imagePullSecrets:` and save it.
+Leave the `uid` value set the same as you found it.
 
-The output of the `sa.yaml` file is similar to this:
+After you've made those changes, the content of your `sa.yaml` will look something like this:
```
Comment (kbhawkey) [inline content/en/docs/tasks/configure-pod-container/configure-service-account.md]: nit: This is the first time `sa.yaml` is mentioned. When is the output saved to sa.yaml?

File: content/en/docs/tasks/configure-pod-container/configure-service-account.md [hunk 16]
```diff
@@ -245,49 +279,44 @@ imagePullSecrets:
...
 
-Finally replace the serviceaccount with the new updated `sa.yaml` file
-
-```shell
-kubectl replace serviceaccount default -f ./sa.yaml
-```
-
-Now, any new pods created in the current namespace will have this added to their spec:
+Now you've made those changes, any new Pods created in the current namespace
+will have this added to their spec:
```
Comment (kbhawkey) [inline content/en/docs/tasks/configure-pod-container/configure-service-account.md]: nit: Do you need: Now that you've made those changes? Above is, `After you've made those changes`. Could just say, `Any new Pods ...`

File: content/en/docs/reference/access-authn-authz/service-accounts-admin.md [hunk 17]
```diff
@@ -4,115 +4,142 @@ reviewers:
...
 
-Support for authorization and user accounts is planned but incomplete.  Sometimes
-incomplete features are referred to in order to better describe service accounts.
+A _ServiceAccount_ provides an identity for processes that run in a Pod.
+
+A process inside a Pod can use the identiy of its associated service account to
+authenticate to the cluster's API server.
+
+For an introduction to service accounts, read [configure service accounts](docs/tasks/configure-pod-container/configure-service-account/).
+
```
Comment (kbhawkey) [inline content/en/docs/reference/access-authn-authz/service-accounts-admin.md]: Pls. fix the link: /docs/tasks/...

File: content/en/docs/reference/access-authn-authz/service-accounts-admin.md (hunk 14 above)
Comment (sftim) [reply content/en/docs/reference/access-authn-authz/service-accounts-admin.md]: Once PR #15374 is ready and merged, I'd like to link to the controller reference information from these service account pages,

**Comment by zacharysarah** [comment]
@sftim 👋 Just checking in. Do you have what you need to continue?

**Comment by sftim** [comment]
Hi @zacharysarah 
Yep, this does need more work but (I hope) not much more, and I've got feedback to draw on.

**Comment by sftim** [comment]
Revised

**Comment by sftim** [comment]
@chenopis or @tengqm - can you review?

File: content/en/docs/tasks/configure-pod-container/configure-service-account.md [hunk 18]
```diff
@@ -3,28 +3,24 @@ reviewers:
...
 - thockin
-title: Configure Service Accounts for Pods
+title: Configure ServiceAccounts for Pods
 content_template: templates/task
 weight: 90
 ---
 
 {{% capture overview %}}
-A service account provides an identity for processes that run in a Pod.
+Kubernetes recognises two distinct ways for clients outside the
```
Comment (kbhawkey) [inline content/en/docs/tasks/configure-pod-container/configure-service-account.md]: hello @sftim . I read through the page. There are a few **minor** nits that could be addressed before merging.
line 12: recognizes 😄 
line 19: spelling `yoy`, recognizes
line 40: capitalize namespace or not?
line 61: the `kubelet`?  enclose in backticks or not?
line 95: possible edit: You can list all ServiceAccount resources in your current namespace with:
line 120: possible edit: You can get a YAML output of ...
line 314: fix the link with forward slash before docs, [projected volume](/docs/concepts/storage/volumes/#projected) type called

**Comment by sftim** [comment]
Thanks for the feedback @kbhawkey 
I think it's all fixed now(?)

**Comment by kbhawkey** [comment]
@sftim , I read through the `Configure ServiceAccounts ...` page again. Some of the _small_ nits remain (spelling:recognises) and there is a link to `automatically` that needs to be corrected. Also check for a typo where you changed the sentence about YAML output.
The second part of this PR, `Managing ServiceAccounts`, is missing the reference information about ServiceAccount automation and Kubernetes controllers. For me, this part of the page is the most [important](https://kubernetes.io/docs/reference/access-authn-authz/service-accounts-admin#service-account-automation). I realize that there are some dependencies with this PR, but perhaps the controller content should stay with the page until the controller reference section is created in another commit (unless the content is entirely outdated).

**Comment by sftim** [comment]
I think what I'll do is close this PR, _keep the branch around_, and try to get these changes in once there are other Controller-related PRs already merged.
Thanks for all feedback so far.

**Pull request closed without merging**